testing/src/main/java/org/conscrypt/MethodFilter.java

```diff
@@ -0,0 +1,216 @@
+package org.conscrypt;
+
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+
+/**
+ * Test support class for filtering collections of {@link Method}.  Each filter is a list of
+ * predicates which must all be true for a Method in order for it to be included it the output.
+ */
+public class MethodFilter {
+    private final String name;
+    private final CompoundMethodPredicate predicates = new CompoundMethodPredicate();
+    private int expectedSize = 0;
+
+    public MethodFilter(String name) {
+        this.name = name;
+    }
+
+    public List<Method> filter(Iterable<Method> input) {
+                List<Method> result = new ArrayList<>();
+        for (Method method : input) {
+            if (method.getParameterTypes().length > 0 && predicates.test(method)) {
```

Comment:
There is dedicated predicate for parameters length. Is it always applied implicitly?

Reply:
This was a mistake :)